test/test_telemetry.py

```diff
@@ -44,3 +62,54 @@ def run(self, add_isolated_node_eval: bool = False):
     # todo replace [1] with .kwargs when moving from python 3.7 to 3.8 in CI
     assert mock_posthog_capture.call_args[1]["event"] == "TestClass.run executed"
     assert mock_posthog_capture.call_args[1]["properties"]["add_isolated_node_eval"]
+
+
+def num_lines(path: Path):
+    if path.is_file():
+        with open(path, "r") as f:
+            return len(f.readlines())
+    return 0
+
+
+@patch("haystack.telemetry.LOG_PATH", Path("~/.haystack/telemetry_test.log").expanduser())
+def test_write_to_file():
```

Comment:
I think this counts as an integration test, so we should mark it with `@pytest.mark.integration`